core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -272,18 +309,24 @@ class CachedDataFrame(df: Dataset[Row], val queryString: String,
       newpls: mutable.ArrayBuffer[ParamLiteral]): Unit = {
     if (allbcplans.nonEmpty && !firstAccess) {
       allbcplans.foreach { case (bchj, refs) =>
+        println(s"Repreparing for bcplan = ${bchj} with new pls = ${newpls.toSet}")
         val broadcastIndex = refs.indexWhere(_.isInstanceOf[Broadcast[_]])
         val newbchj = bchj.transformAllExpressions {
           case pl@ParamLiteral(_, _, p) =>
             val np = newpls.find(_.pos == p).getOrElse(pl)
-            ParamLiteral(np.value, np.dataType, p)
+            val x = ParamLiteral(np.value, np.dataType, p)
+            x.considerUnequal = true
```

Comment:
Right have mentioned the same above which should be done if we are not going to tackle tokenization for bchj soon.